### PR TITLE
Bump to build with go1.18 (#19120 et al)

### PR DESCRIPTION
**Pull request by techknowlogick**
Backport #19120 
Backport #19099 
Backport #18874 
Backport #18420
Backport #19128
Backport #18270 

Bump to build with go1.18

### CODE REVIEW

File: modules/util/net_test.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+		},
+		// case 3
+		{
+			ip:        "8.8.8.8",
+			isPrivate: false,
+		},
+		// case 4
+		{
+			ip:        "::1",
+			isPrivate: false, // TODO: according to go, this isn't private?
```
Comment (Gusted) [inline modules/util/net_test.go]: Private != Local IP's, See: https://datatracker.ietf.org/doc/html/rfc1918#section-3

File: modules/util/net_test.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		},
+		// case 4
```
Comment (6543) [inline modules/util/net_test.go]:
```suggestion
```

Comment (Gusted) [reply modules/util/net_test.go]: Might want to frontport the removals of the TODO's?

Comment (6543) [reply modules/util/net_test.go]: more ...

**Review by silverwind — APPROVED** [review]

File: modules/util/net_test.go (hunk 2 above)
Comment (6543) [reply modules/util/net_test.go]: -> #19128

**Comment by techknowlogick** [comment]
I think some go.mod changes I made are causing this to fail:
```
--- FAIL: TestFile (0.02s)

    --- FAIL: TestFile/.drone.yml (0.01s)

        highlight_test.go:100: File() = [<span class="line"><span class="cl"><span class="nt">kind</span><span class="p">:</span><span class="w"> </span><span class="l">pipeline</span> </span></span><span class="line"><span class="cl"><span class="w"></span><span class="nt">name</span><span class="p">:</span><span class="w"> </span><span class="l">default</span> </span></span><span class="line"><span class="cl"> </span></span><span class="line"><span class="cl"><span class="w"></span><span class="nt">steps</span><span class="p">:</span> </span></span><span class="line"><span class="cl"><span class="w"></span>- <span class="nt">name</span><span class="p">:</span><span class="w"> </span><span class="l">test</span> </span></span><span class="line"><span class="cl"><span class="w">	</span><span class="nt">image</span><span class="p">:</span><span class="w"> </span><span class="l">golang:1.13</span> </span></span><span class="line"><span class="cl"><span class="w">	</span><span class="nt">environment</span><span class="p">:</span> </span></span><span class="line"><span class="cl"><span class="w"></span><span class="w">		</span><span class="nt">GOPROXY</span><span class="p">:</span><span class="w"> </span><span class="l">https://goproxy.cn</span> </span></span><span class="line"><span class="cl"><span class="w">	</span><span class="nt">commands</span><span class="p">:</span> </span></span><span class="line"><span class="cl"><span class="w"></span><span class="w">	</span>- <span class="l">go get -u</span> </span></span><span class="line"><span class="cl"><span class="w">	</span>- <span class="l">go build -v</span> </span></span><span class="line"><span class="cl"><span class="w">	</span>- <span class="l">go test -v -race -coverprofile=coverage.txt -covermode=atomic</span><span class="w">

            </span></span></span> <span class="w">

            </span>], want [<span class="nt">kind</span><span class="p">:</span><span class="w"> </span><span class="l">pipeline</span> <span class="w"></span><span class="nt">name</span><span class="p">:</span><span class="w"> </span><span class="l">default</span> <span class="w">

            </span> <span class="w"></span><span class="nt">steps</span><span class="p">:</span> <span class="w"></span>- <span class="nt">name</span><span class="p">:</span><span class="w"> </span><span class="l">test</span> <span class="w">	</span><span class="nt">image</span><span class="p">:</span><span class="w"> </span><span class="l">golang:1.13</span> <span class="w">	</span><span class="nt">environment</span><span class="p">:</span> <span class="w"></span><span class="w">		</span><span class="nt">GOPROXY</span><span class="p">:</span><span class="w"> </span><span class="l">https://goproxy.cn</span> <span class="w">	</span><span class="nt">commands</span><span class="p">:</span> <span class="w"></span><span class="w">	</span>- <span class="l">go get -u</span> <span class="w">	</span>- <span class="l">go build -v</span> <span class="w">	</span>- <span class="l">go test -v -race -coverprofile=coverage.txt -covermode=atomic</span><span class="w">

            </span> <span class="w">

            </span>]

    --- FAIL: TestFile/.drone.yml_-_trailing_space (0.01s)

        highlight_test.go:100: File() = [<span class="line"><span class="cl"><span class="nt">kind</span><span class="p">:</span><span class="w"> </span><span class="l">pipeline</span> </span></span><span class="line"><span class="cl"><span class="w"></span><span class="nt">name</span><span class="p">:</span><span class="w"> </span><span class="l">default  </span> </span></span><span class="line"><span class="cl"> </span></span><span class="line"><span class="cl"><span class="w"></span><span class="nt">steps</span><span class="p">:</span> </span></span><span class="line"><span class="cl"><span class="w"></span>- <span class="nt">name</span><span class="p">:</span><span class="w"> </span><span class="l">test</span> </span></span><span class="line"><span class="cl"><span class="w">	</span><span class="nt">image</span><span class="p">:</span><span class="w"> </span><span class="l">golang:1.13</span> </span></span><span class="line"><span class="cl"><span class="w">	</span><span class="nt">environment</span><span class="p">:</span> </span></span><span class="line"><span class="cl"><span class="w"></span><span class="w">		</span><span class="nt">GOPROXY</span><span class="p">:</span><span class="w"> </span><span class="l">https://goproxy.cn</span> </span></span><span class="line"><span class="cl"><span class="w">	</span><span class="nt">commands</span><span class="p">:</span> </span></span><span class="line"><span class="cl"><span class="w"></span><span class="w">	</span>- <span class="l">go get -u</span> </span></span><span class="line"><span class="cl"><span class="w">	</span>- <span class="l">go build -v</span> </span></span><span class="line"><span class="cl"><span class="w">	</span>- <span class="l">go test -v -race -coverprofile=coverage.txt -covermode=atomic</span> </span></span><span class="line"><span class="cl"><span class="w">	</span></span></span>], want [<span class="nt">kind</span><span class="p">:</span><span class="w"> </span><span class="l">pipeline</span> <span class="w"></span><span class="nt">name</span><span class="p">:</span><span class="w"> </span><span class="l">default  </span> <span class="w">

            </span> <span class="w"></span><span class="nt">steps</span><span class="p">:</span> <span class="w"></span>- <span class="nt">name</span><span class="p">:</span><span class="w"> </span><span class="l">test</span> <span class="w">	</span><span class="nt">image</span><span class="p">:</span><span class="w"> </span><span class="l">golang:1.13</span> <span class="w">	</span><span class="nt">environment</span><span class="p">:</span> <span class="w"></span><span class="w">		</span><span class="nt">GOPROXY</span><span class="p">:</span><span class="w"> </span><span class="l">https://goproxy.cn</span> <span class="w">	</span><span class="nt">commands</span><span class="p">:</span> <span class="w"></span><span class="w">	</span>- <span class="l">go get -u</span> <span class="w">	</span>- <span class="l">go build -v</span> <span class="w">	</span>- <span class="l">go test -v -race -coverprofile=coverage.txt -covermode=atomic</span> <span class="w">	</span>]

FAIL
```

**Review by 6543 — APPROVED** [review]

**Comment by zeripath** [comment]
> 
> ```
> --- FAIL: TestFile (0.02s)
> 
>     --- FAIL: TestFile/.drone.yml (0.01s)
> 
>         highlight_test.go:100: File() = [<span class="line"><span class="cl"><span class="nt">kind</span><span class="p">:</span><span class="w"> </span><span class="l">pipeline</span> </span></span><span class="line"><span class="cl"><span class="w"></span><span class="nt">name</span><span class="p">:</span><span class="w"> </span><span class="l">default</span> </span></span><span class="line"><span class="cl"> </span></span><span class="line"><span class="cl"><span class="w"></span><span class="nt">steps</span><span class="p">:</span> </span></span><span class="line"><span class="cl"><span class="w"></span>- <span class="nt">name</span><span class="p">:</span><span class="w"> </span><span class="l">test</span> </span></span><span class="line"><span class="cl"><span class="w">	</span><span class="nt">image</span><span class="p">:</span><span class="w"> </span><span class="l">golang:1.13</span> </span></span><span class="line"><span class="cl"><span class="w">	</span><span class="nt">environment</span><span class="p">:</span> </span></span><span class="line"><span class="cl"><span class="w"></span><span class="w">		</span><span class="nt">GOPROXY</span><span class="p">:</span><span class="w"> </span><span class="l">https://goproxy.cn</span> </span></span><span class="line"><span class="cl"><span class="w">	</span><span class="nt">commands</span><span class="p">:</span> </span></span><span class="line"><span class="cl"><span class="w"></span><span class="w">	</span>- <span class="l">go get -u</span> </span></span><span class="line"><span class="cl"><span class="w">	</span>- <span class="l">go build -v</span> </span></span><span class="line"><span class="cl"><span class="w">	</span>- <span class="l">go test -v -race -coverprofile=coverage.txt -covermode=atomic</span><span class="w">
> ...
> ```

These are related to upgrading goldmark...

**Comment by 6543** [comment]
🚀

**Comment by 6543** [comment]
now we have merged v1.17.0-dev into v1.16 ;)

... joke

**Pull request merged**